<a href="https://colab.research.google.com/github/fadhlijahfal1/SD2104-Pemrograman-Lanjut-/blob/main/For_While_Fadhli_2C2230008.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**🖥️👩🏻‍💻PRAKTIKUM PEMROGRAMAN LANJUT🔥**
##**Pertemuan (2) PENGULANGAN (LOOP) - Senin, 21 Oktober 2024**

👤 Nama: Fadhli Jahfal Aufa Maulana / NRP: 2C2230008 / Kelas: III/A Sains Data

💬Deskripsi Pembahasan Materi Pengulangan:

1.1.  Pengulangan (LOOP) 🔂
- While Loop 🐋

- For Loop

1.2. Pengulangan Bersarang ( NESTED LOOP) 🔂🐦‍




#🤸🏻‍♀️ EKSPLORASI PRIBADI DI KELAS ✨

In [ ]:
#List
my_list = [1, 2, 3, 4, 5]
my_list[0] = 10 #List dapat diubah
print(my_list) #Hasil outout [10, 2, 3, 4, 5]

[10, 2, 3, 4, 5]


In [ ]:
#Tuple
my_tuple = (1, 2, 3, 4, 5)
print(my_tuple) #Tuple tidak bisa di ubah urutannya

(1, 2, 3, 4, 5)


**Pengulangan For Loop**

In [ ]:
#Pengulangan For Loop
var_list = [1,2,3,4,5]
for i in var_list:
  print("i = ", i)

i =  1
i =  2
i =  3
i =  4
i =  5


In [ ]:
huruf = "Fadhli Jahfal"
for i in huruf:
  print(i)

F
a
d
h
l
i
 
J
a
h
f
a
l


In [ ]:
for i in range(10): #(0,10,1) ARTINYA (START,STOP,STEP)
  print(i)

0
1
2
3
4
5
6
7
8
9


In [ ]:
#Perulangan dengan range yang benar
for i in range(1,10,2):
  print(i)

1
3
5
7
9


In [ ]:
for i in range (1000):
  print("Hello World")

In [ ]:
for i in range (2, 5, 1):
  print(i)

2
3
4


In [ ]:
for i in range (7, 2, -1):
  print(i)

7
6
5
4
3


In [ ]:
a = int(input())
b = int(input())

for i in range (a, b+1):
  print(i)

**Pengulangan While Loop**

In [ ]:
counter = 1
while counter <= 5:
  print("counter = ", counter)
  counter += 1

counter =  1
counter =  2
counter =  3
counter =  4
counter =  5


In [ ]:
a = int(input())
b = int(input())
i = a
while (i <= b):
  print(i)
  i += 2

**Nested Loop**

In [ ]:
n = int(input())

for i in range(n):
  for j in range(n):
    print("*", end="")

5
*************************

**Pembahasan Soal**

In [ ]:
#Kelipatan Terdekat

N = int(input("Masukkan N : "))

i = N + 1
while i % 5 != 0:
  i += 1

print("Output: ", i)

In [ ]:
#Segitiga Bintang

N = int(input("Masukkan N : "))

for i in range (1, N+1):
  print("*" * i)

In [ ]:
# Urutan Kebawah

N = int(input("Masukkan N : "))

for i in range (N, 0, -1):
  for i in range (1, i+1):
    print(i, end="")
  print()

In [ ]:
# Kelipatan 2&3

N = int(input("Masukkan N : "))

#Loop dari 1 hingga N
for i in range (1, N+1):
  if i % 2 == 0 and i % 3 == 0:
    print("DuaTiga", end=" ")
  elif i % 2 == 0:
    print("Dua", end=" ")
  elif i % 3 == 0:
    print("Tiga", end=" ")
  else:
    print(i, end="")
print()

- Interpretasi coding:

- Interpretasi Hasil:

#🧠 LATIHAN SOAL DI KELAS ⏱️

**Problem (1) Membuat program menerima bilangan N dan menuliskan bilangan 10^x terkecil yang lebih dari N**

In [ ]:
N = int(input("Masukkan N : "))
pangkat = 0

while 10 ** pangkat <= N:
  pangkat += 1

print("Output: ", 10 ** pangkat)

Masukkan N : 1234
Output:  10000


**Interpretasi coding:**
1. Input program diatas untuk memasukkan nilai N
2. variabel "pangkat" diinisialisasi dengan 0
3. perulangan menggunakan while dikarenakan kondisi yang tidak pasti. artinya
   program tersebut tidak tahu sebelumnya berapa kali perulangan harus dilakukan
4. Perulangan akan terus berjalan selama hasil dari 10 pangkat masih lebih  
   kecil dari N. ketika sudah melewati atau sama dengan N, perulangan akan berhenti.
**Interpretasi Hasil:**

Masukkan N : 5

output : 10

Masukkan N : 1234

output : 10000

**Problem (2) Membuat program menerima bilangan N dan membuat pola segitiga angka dengan dasar N.**

In [ ]:
N = int(input("Masukkan N: "))

for i in range(1, N + 1): #Loop luar baris
    for j in range(1, i + 1): #Loop dalam baris
        print(j, end=" ")
    print()

Masukkan N: 5
1 
1 2 
1 2 3 
1 2 3 4 
1 2 3 4 5 


**Interpretasi coding:**
1. Pada program ini, kita menggunakan nested loop karena pada output yang
   menghasilkan segitiga angka memiliki jumlah angka yang bertambah 1.
2. Nested loop dapat mengatur jumlah pengulangan pada setiap baris
3. Loop luar baris berguna untuk mengulang baris 1 hingga baris N
4. Loop dalam baris mengulang dari kolom 1 hingga kolom i
5. print (j, end="") berguna untuk mencetak angka j tanpa pindah baris
6. print() berguna untuk pindah ke baris baru setelah setiap baris selesai
   dicetak

**Interpretasi Hasil:**

*1 *1 2 *1 2 3 *1 2 3 4 *1 2 3 4 5

#🏆 POST TEST 🎯

**Problem (1) membuat program jika "x" kelipatan dari a maka keluarkan "Siap". Jika "x" kelipatan dari b maka keluarkan "bang". jika "x" kelipatan c maka keluarkan "jago". jika "x" bukan merupakan kelipatan seluruhnya "else", maka tampilkan angka aja.**

In [ ]:
N = int(input("Masukkan N : "))
a = int(input("Masukkan A : "))
b = int(input("Masukkan B : "))
c = int(input("Masukkan C : "))

for x in range (1, N+1):
  if x % a == 0 and x % b == 0 and x % c == 0:
    print("SiapBangJago", end=" ")
  elif x % a == 0 and x % b == 0:
    print("SiapBang", end=" ")
  elif x % a == 0 and x % c == 0:
    print("SiapJago", end=" ")
  elif x % b == 0 and x % c == 0:
    print("BangJago", end=" ")
  elif x % a == 0:
    print("Siap", end=" ")
  elif x % b == 0:
    print("Bang", end=" ")
  elif x % c == 0:
    print("Jago", end=" ")
  else:
    print(x, end=" ")
print()

Masukkan N : 12
Masukkan A : 2
Masukkan B : 3
Masukkan C : 4
1 Siap Bang SiapJago 5 SiapBang 7 SiapJago Bang Siap 11 SiapBangJago 


**Interpretasi coding:**

1. Input program N, a, b, c
2. Menggunakan pengulangan for loop untuk mengulang angka dari 1 hingga N
3. pada saat menggunakan if elif else, kita menjabarkan dari tiap masing masing
   angka, apabila kelipatan x habis dibagi a, b, c maka akan keluar output "SiapBangJago"
4. Apabila kelipatan x habis dibagi a dan b maka akan keluar output "SiapBang"
5. Apabila kelipatan x habis dibagi b dan c maka akan keluar output "BangJago"
6. Apabila kelipatan x habis dibagi a maka akan keluar output "Siap"
7. Apabila kelipatan x habis dibagi b maka akan keluar output "Bang"
8. Apabila kelipatan x habis dibagi c maka akan keluar output "Jago"
9. Apabila kelipatan x tidak dapat dibagi a,b,c maka akan keluar output  
   bilangan itu sendiri "N"

**Interpretasi hasil:**

Masukkan nilai N = 12
Masukkan Nilai a = 2
Masukkan Nilai b = 3
Masukkan Nilai c = 4

Output : 1 Siap Bang SiapJago 5 SiapBang 7 SiapJago Bang Siap 11 SiapBangJago

📌Catatan:
  - Notebook python ini dibuat untuk memudahkan temen-temen belajar ngoding dan pengumpulan tugas.
  - Silakan download notebook ini lalu upload di Google Collab milik kalian agar format `pertemuan`, `topik pertemuan`,
`tanggal`, `deskripsi pembahasan materi`, `baris coding` dan `interpretasi` bisa disesuaikan dengan kebutuhan teman-teman.

  - Interpretasi coding dan hasil dimaksudkan agar teman-teman semakin memahami apa yang dipelajari.

  - Deskripsi pembahasan materi dimaksudkan agar teman-teman semakin memahami apa fokus yang dipelajari.

  - Tulisan di setiap sebelum baris code disebut `markdown`. Ada beberapa jenis simbol yang memudahkan kepenulisan `markdown` kita, yaitu seperti : untuk mengatur ukuran tulisan biasanya tulisan pada header menggunakan ini`#`, `##`, `###`, untuk membuat pointer menggunakan ini `-`, untuk membuat tulisan ter-highlight menggunakan ini *` `* , untuk short-cut membuat tulisan jadi **bold** gunakan ini `** **`, dan untuk *tulisan miring* gunakan ini `* *` .

  - Semangat belajar! 🌟✨

©️ LABORATORIUM SAINS DATA 2024